### PR TITLE
better docs

### PR DESCRIPTION
**Pull request by prateekbh**
R: @jeffposnick @addyosmani @gauntface

This adds better documentation to background-sync

P.S. it breaks a lint which I am not sure how to fix

### CODE REVIEW

File: packages/workbox-background-sync/demo/sw.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /* global workbox*/
 importScripts(
-	'../build/background-sync-queue.js',
-	'../../workbox-routing/build/workbox-routing.js',
-	'../../workbox-runtime-caching/build/workbox-runtime-caching.js'
+	'../build/importScripts/workbox-background-sync.dev.v0.0.1.js',
+	'../../workbox-routing/build/importScripts/workbox-routing.dev.v0.0.1.js',
+	'../../workbox-runtime-caching/build/importScripts/workbox-runtime-caching.dev.v0.0.1.js'
```
Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: @gauntface Noob question: this path itself is longer than 80 chars, and fails lint how to fix this?

Comment (gauntface) [inline packages/workbox-background-sync/demo/sw.js]: 'sdfsdfsdf' +
'sdsdfsdfsdf'

Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: Fixed! and feeling super noob 😬😬😬

File: packages/workbox-background-sync/demo/sw.js [hunk 2]
```diff
@@ -1,35 +1,36 @@
 /* global workbox*/
 importScripts(
-	'../build/background-sync-queue.js',
-	'../../workbox-routing/build/workbox-routing.js',
-	'../../workbox-runtime-caching/build/workbox-runtime-caching.js'
+  '../build/importScripts/workbox-background-sync.dev.v0.0.1.js',
```
Comment (jeffposnick) [inline packages/workbox-background-sync/demo/sw.js]: You should be able to load this in via `/__test/bundle/<pkg>` if you assume that these demos will always be run when served by our test server.

File: packages/workbox-background-sync/demo/sw.js [hunk 3]
```diff
@@ -1,35 +1,36 @@
...
-			self.registration.showNotification('Background sync demo', {
-				body: 'Product has been purchased.',
-				icon: 'https://shop.polymer-project.org/images/shop-icon-384.png',
-			});
-		},
-		onRetryFailure: (hash) => {},
-	},
-	dbName: 'queues',
+  {
+    onResponse: async(hash, res) => {
```
Comment (jeffposnick) [inline packages/workbox-background-sync/demo/sw.js]: What's `hash` and `res` used for? Could you use them in the sample code, or else add a comment explaining what it's for?

File: packages/workbox-background-sync/demo/sw.js [hunk 4]
```diff
@@ -1,35 +1,36 @@
...
+      self.registration.showNotification('Background sync demo', {
+        body: 'Product has been purchased.',
+        icon: 'https://shop.polymer-project.org/images/shop-icon-384.png',
+      });
+    },
+    onRetryFailure: (hash) => {},
```
Comment (jeffposnick) [inline packages/workbox-background-sync/demo/sw.js]: Can you include some code with a common use case?

File: packages/workbox-background-sync/demo/sw.js (hunk 3 above)
Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: done! turned this into a more generic example

File: packages/workbox-background-sync/demo/sw.js (hunk 4 above)
Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: common use case?
although i have made this code itself pretty generic now, is anything else needed?

File: packages/workbox-background-sync/demo/sw.js (hunk 2 above)
Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: done! thanks for the pointer

File: packages/workbox-background-sync/demo/sw.js (hunk 3 above)
Comment (prateekbh) [inline packages/workbox-background-sync/demo/sw.js]: also changed these to friendlier names

File: packages/workbox-background-sync/demo/index.html [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <html>
    <head>
-      <title>Riot code splitting</title>
+      <title>Background sync queue demo</title>
```
Comment (jpmedley) [inline packages/workbox-background-sync/demo/index.html]: Should 'Background Sync Queue Demo'. (Title case)

**Comment by prateekbh** [comment]
@jpmedley fixed the title case

**Pull request merged**